### ISSUE TITLE
False positive in Layout/EndAlignment for start_of_line style

### ISSUE THREAD

**Issue by jonas054** [bbatsov/rubocop#5582]
Here's a problem I found. I've made a fix myself. A PR will be submitted soon.

## Expected behavior

No offense reported for the file.

## Actual behavior

With the configuration
```yaml
# .rubocop.yml
Layout/EndAlignment:
  EnforcedStyleAlignWith: start_of_line
  AutoCorrect: true
```

and the file
```ruby
# file.rb
var =
  if test
  end
```

the command
```
rubocop --auto-correct
```

reports
```
Inspecting 1 file
W

Offenses:

file.rb:3:3: W: [Corrected] Layout/EndAlignment: end at 4, 2 is not aligned with var = at 2, 0.
  end
  ^^^

1 file inspected, 1 offense detected, 1 offense corrected
```

There are two problems with this.
1) `var =` is not at the start of the line where `if` is, so `end` should not be aligned with it.
2) It says `[Corrected]` but no correction is made.

## Steps to reproduce the problem

Run `rubocop -a` in a directory with `file.rb` and `.rubocop.yml` as described.

## RuboCop version

```
$ rubocop -V
0.52.1 (using Parser 2.5.0.0, running on ruby 2.5.0 x86_64-linux)
```
